vision/google/cloud/vision/image.py

```diff
@@ -30,22 +30,35 @@ class Image(object):
     :type content: bytes
     :param content: Byte stream of an image.
 
+    :type filename: str
+    :param filename: Filename to image.
+
     :type source_uri: str
     :param source_uri: Google Cloud Storage URI of image.
 
     :type client: :class:`~google.cloud.vision.client.Client`
     :param client: Instance of Vision client.
     """
 
-    def __init__(self, client, content=None, source_uri=None):
+    def __init__(self, client, content=None, filename=None, source_uri=None):
+        sources = set(source for source in (content, filename, source_uri)
+                      if source is not None)
+        if len(sources) != 1:
+            raise ValueError(
+                'Specify exactly one of \'content\', \'filename\', or'
+                '\'source_uri\'.')
+
         self.client = client
-        self._content = None
-        self._source = None
 
-        if source_uri:
-            self._source = source_uri
-        else:
-            self._content = _bytes_to_unicode(b64encode(_to_bytes(content)))
+        if filename is not None:
+            with open(filename, 'rb') as file_obj:
+                content = file_obj.read()
```

Comment:
Since `filename` is a helper, I just grab the data from the file and put into `content`.
Then `content` is used as it normally would.

So if you pass in an empty file, that is when `content` would be `None`.